src/OpenSearch/Connections/Connection.php

```diff
@@ -748,14 +748,9 @@ private function tryDeserializeError(array $response, string $errorClass): OpenS
             // 2.0 structured exceptions
             if (is_array($error['error']) && array_key_exists('reason', $error['error']) === true) {
                 // Try to use root cause first (only grabs the first root cause)
-                $root = $error['error']['root_cause'];
-                if (isset($root) && isset($root[0])) {
-                    $cause = $root[0]['reason'];
-                    $type = $root[0]['type'];
-                } else {
-                    $cause = $error['error']['reason'];
-                    $type = $error['error']['type'];
-                }
+                $info = $error['error']['root_cause'][0] ?? $error['error'];
```

Comment:
The previous code was also checking that `$error['error']['root_cause']` was a thing. I think that's good defensive code, I would add a test that trips that and re-add a check here?

Reply:
The `??` works just like the `isset()` language construct and checks all “levels” of the first expression for whether they exist. Checking for the existence of `$error['error']['root_cause']` explicitly is therefore unnecessary. See the docs on [the `??` operator](https://www.php.net/manual/en/language.operators.comparison.php#language.operators.comparison.coalesce) and [`isset()`](https://www.php.net/manual/en/function.isset.php) for details, or run the following:
```
$a = NULL;
$b = $a['foo']['bar']['baz'][0]->foobar['thingie'] ?? 'All fine';
echo $b;
```

Also, you can see in my test that I already do pass a response where `$error['error']['root_cause']` is unset. So, if there was anything to trip, it would already be tripped by that test.